### ISSUE TITLE
Silence expected console errors during E2E tests

### ISSUE THREAD

**Issue by tofumatt** [google/site-kit-wp#842]
Bug Description

The E2E tests produce a lot of "junk" output when running because they have intentional/expected errors that result in `console.error` calls. Jest forwards all console output to the terminal, which results in really noisy test runs where things _look_ like they're not working when in fact they are (see screenshot below).

We should silence expected console output (such as `Failed to load resource: the server responded with a status`) so only relevant errors make it to the console when running the tests.

## Steps to reproduce

1. Run the E2E tests (`npm run test:e2e`)
2. Observe noisy console output

## Screenshots

<img width="673" alt="Screenshot 2019-11-12 at 17 36 18" src="https://user-images.githubusercontent.com/90871/68695436-f26f1c00-0572-11ea-9f24-9e1991c8b1ab.png">

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

* All e2e tests that result in error logs in the console should be updated so that the console is muted for the specific test.
* The console should _not_ be muted generally throughout all tests, as that would increase our chance of missing problems in a test. It therefore should be done on case-by-case basis.

## Implementation Brief

* Add `'@wordpress/jest-console'` into the `setupFilesAfterEnv` array in `tests/e2e/jest.config.js`
* Update all debugging `console` calls in `config/bootstrap.js` to use `console.debug` instead
* Update E2E tests with the necessary assertions to `expect( console ).` `toHaveErrored`, `toHaveWarned`, `toHaveLogged`.
	- Any of the above messages which are not necessarily expected, but are not relevant can be filtered out and ignored by adding cases for them in the `observeConsoleLogging` function in `bootstrap.js`  
	E.g. React notices, cookie notices, or anything which is more/less unrelated to our tests ([one example where this may be relevant for WP](https://github.com/WordPress/gutenberg/blob/5ced7f1ab18d6019da01f2cb812697ddde8837b7/packages/e2e-tests/config/setup-test-framework.js#L139-L145))

## QA Brief

* E2E tests should produce no extra output when run as shown above (unless there is a legitimate error or failed assertion)

## Changelog entry

* N/A

**Comment by felixarntz** [comment]
@tofumatt I believe this is no longer relevant after #1770? I'll close this for now, but feel free to reopen if I missed something.

**Comment by felixarntz** [comment]
Oh I just noticed this about e2e tests 😆

**Comment by felixarntz** [comment]
IB ✅

**Comment by felixarntz** [comment]
## QA ❌ 

@aaemnnosttv @ivankruchkoff I believe the changes here broke our expected feature flags behavior. For example I no longer see the new widgets when doing a development build with `npm run build:dev`.

Possibly caused by https://github.com/google/site-kit-wp/commit/a43d3cf62887a664487cb04ffeddd7823a7cb39a?

**Comment by aaemnnosttv** [comment]
@felixarntz yeah that's the culprit – fix incoming.

**Comment by tofumatt** [comment]
I see this as working on CI, but for me, locally, I'm seeing:

<img width="707" alt="Screenshot 2020-10-30 at 19 52 08" src="https://user-images.githubusercontent.com/90871/97750802-663e6780-1ae9-11eb-84a5-2fcebdc666d2.png">

I'm not sure if this is timeout or some other setup-related thing, and it seems recent, but unrelated to this change.

If someone else can QA this we'll chalk it up to something being off with my local setup, though what I'm not sure. Un-assigning myself for now…

**Comment by tofumatt** [comment]
Whoops, ran this again with `npm run build:test` and saw a QUIET OUTPUT YAY 🎉 

QA ✅
